contracts/TokenConversionManager.sol

```diff
@@ -135,11 +161,20 @@ contract TokenConversionManager is Ownable, ReentrancyGuard {
         // Mint the tokens and transfer to the User Wallet using the Call function
         // token.mint(amount, msg.sender);
 
-        (bool success, ) = address(token).call(abi.encodeWithSelector(MINT_SELECTOR, to, amount));
+        // Compute the Fee - The fee amount will be subtracted during the transfer
+        uint256 feeAmount = amount.mul(feePercentage).div(10000);
+
+        (bool success, ) = address(token).call(abi.encodeWithSelector(MINT_SELECTOR, address(this), amount));
 
         // In case if the mint call fails
         require(success, "ConversionIn Failed");
 
+        // Do the transfers
+        require(token.transfer(to, amount.sub(feeAmount)), "Unable to transfer token");
```

Comment:
If feePercentage is 100% then we will be transfering 0 right?

Reply:
Yes zero will be transferred. Which is not going to happen as the fee will never be 100%.